### PR TITLE
New Homescreen (Initial change)

### PR DESCRIPTION
**Pull request by microsoftsam**
This is an initial attempt and refactoring the projects menu to show a home screen with carousels for users projects, tutorials, examples and videos.

### CODE REVIEW

File: localtypings/react-slick.d.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/// <reference path="../typings/globals/react/index.d.ts" />
```
Comment (pelikhan) [inline localtypings/react-slick.d.ts]: Still needed?

File: theme/common.less [hunk 2]
```diff
@@ -13,6 +13,11 @@ html, body {
     min-width: 320px;
 }
 
+* {
```
Comment (pelikhan) [inline theme/common.less]: why do we need this?

File: theme/themes/pxt/views/card.overrides [hunk 3]
```diff
@@ -5,4 +5,77 @@
 .ui.card .ui.button.massive {
     border-bottom-left-radius: 0;
     border-bottom-right-radius: 0;
+}
+
+/* File card for the Home screen */
+.ui.card.file {
```
Comment (pelikhan) [inline theme/themes/pxt/views/card.overrides]: File card? What files are we linking to?

File: webapp/src/app.tsx [hunk 4]
```diff
@@ -1631,8 +1635,7 @@ ${compileService && compileService.githubCorePackage && compileService.gittag ?
         const tutorialOptions = this.state.tutorialOptions;
         const inTutorial = !!tutorialOptions && !!tutorialOptions.tutorial;
         const docMenu = targetTheme.docMenu && targetTheme.docMenu.length && !sandbox && !inTutorial;
-        const gettingStarted = !sandbox && !inTutorial && !this.state.sideDocsLoadUrl && targetTheme && targetTheme.sideDoc && isBlocks && !targetTheme.useStartPage;
-        const gettingStartedTooltip = lf("Open beginner tutorial");
+        const gettingStarted = !sandbox && !inTutorial && !this.state.sideDocsLoadUrl && targetTheme && targetTheme.sideDoc && isBlocks;
```
Comment (pelikhan) [inline webapp/src/app.tsx]: The gettingStarted button goes away right?

File: localtypings/react-slick.d.ts (hunk 1 above)
Comment (microsoftsam) [reply localtypings/react-slick.d.ts]: Yes. For slider. Once @riknoll's carousel PR rolls in, we can remove. This PR depends on react-slick.

File: theme/common.less (hunk 2 above)
Comment (microsoftsam) [reply theme/common.less]: For flexbox support with react-slick. see README at: https://github.com/akiran/react-slick

File: theme/themes/pxt/views/card.overrides (hunk 3 above)
Comment (microsoftsam) [reply theme/themes/pxt/views/card.overrides]: I refer to header as file cards.

File: webapp/src/app.tsx (hunk 4 above)
Comment (microsoftsam) [reply webapp/src/app.tsx]: Yes no getting started button anymore. However, there is a getting started button on the home screen, and so I'm re-using the logic to determine when that shows.

File: localtypings/pxtpackage.d.ts [hunk 5]
```diff
@@ -82,7 +82,7 @@ declare namespace pxt {
         time?: number;
         url?: string;
         responsive?: boolean;
-        cardType?: "example" | "codeExample" | "tutorial" | "project";
+        cardType?: "file" | "example" | "codeExample" | "tutorial" | "project";
```
Comment (pelikhan) [inline localtypings/pxtpackage.d.ts]: why are we supporting a file type?

File: theme/themes/pxt/views/card.overrides (hunk 3 above)
Comment (pelikhan) [reply theme/themes/pxt/views/card.overrides]: what is a `header`?

File: webapp/src/app.tsx (hunk 4 above)
Comment (pelikhan) [reply webapp/src/app.tsx]: On the home screen, getting started button should always be visible.

File: webapp/src/codecard.tsx [hunk 6]
```diff
@@ -44,26 +48,26 @@ export class CodeCardView extends React.Component<pxt.CodeCard, CodeCardState> {
...
                     {card.shortName || card.name ? <div className="header">{card.shortName || card.name}</div> : null}
-                    {card.time ? <div className="meta tall">
-                        {card.time ? <span key="date" className="date">{pxt.Util.timeSince(card.time) }</span> : null}
-                    </div> : undefined}
-                    {card.description ? <div className="description tall">{renderMd(card.description)}</div> : null}
+                    {card.description ? <div className="description tall">{renderMd(card.description) }</div> : null}
                 </div> : undefined }
+                {card.time ? <div className="meta">
+                    {card.time ? <span key="date" className="date">{pxt.Util.timeSince(card.time) }</span> : null}
+                </div> : undefined}
```
Comment (pelikhan) [inline webapp/src/codecard.tsx]: changes to the codecard rendering should be also reflected in docsrender.ts

**Comment by pelikhan** [comment]
LGTM. Great start. Let's get it in and polish as the new carousel comes alive.

File: theme/themes/pxt/views/card.overrides (hunk 3 above)
Comment (microsoftsam) [reply theme/themes/pxt/views/card.overrides]: a local project.

File: webapp/src/app.tsx (hunk 4 above)
Comment (microsoftsam) [reply webapp/src/app.tsx]: Okay.

**Comment by pelikhan** [comment]
@microsoftsam can we get this in?

**Comment by microsoftsam** [comment]
Just need to look into your docrender comment. I haven't found the code you're referring to yet

**Comment by pelikhan** [comment]
pxt.docs.codeCard.render

**Pull request merged**